### DISCUSSION TITLE
Help setting a 2-Axes lathe

### DISCUSSION THREAD

**Discussion by mvackel** [Paciente8159/uCNC#517] [Config and issues] [answered]
I'm trying to config a lathe (2-axes, Z along the spindler and X). The axes positions and names seems to be important - at least I couldn't make Fusion360 use other options. I've searched for information about this but couldn't find anything that really helps. 
Using GRBL I thought that I could configure a 3 axis machine and change the Homing Cycle, disabling the Y homing and simply ignoring the Y motor, disconnecting it from the board. Using uCNC I couldn't find the Homing Cycle compilation settings that are present in GRBL's config.h file. Is it possible to change the Homing Cycle in uCNC? How can I do this?

Anyway, maybe I'm missing something because it shouldn't be so hard to configure a ZX Lathe machine, following the conventions. 

Any help will be welcome. 

EDIT: just found the "#define DISABLE_Y_HOMING" line in "cnc_hal_config.h" and it's working, although UGS shows the Y axis trying to home - but it works. Is this the correct approach? And,, again, are there options in uCNC to configure the Home Cycle (order, axes, etc) as in grbl?

**Comment by Paciente8159** [comment, marked as answer]
Sorry for the late reply. I did not notice this topic.

There are multiple ways to kind of do this.

Please check [This](https://github.com/Paciente8159/uCNC/commit/748770113d53d2bc2fa514b99924bfc53994b181)

This is a modification I added to v1.8 to alias Y axis as Z axis in 2 axis machines. Just be aware that you still configure both x and y axis. Z commands will just be aliased internally to use y axis.

On current v1.7 (and 1.8 too) you can modify the homing axis order by changing the order in which each axis is homed. This is set in each kinematic file (for example kinematic for cartesian homing function)
Disabling an axis homing is also an option. You can couple that with changing the axis indexes in the generic kinematicdefs.h

**Comment by zenggood** [comment]
Hello, I would like to swap the aliases of the Z axis and the A axis. Can you help me with that?

**Reply by Paciente8159** [reply]
It's possible to swap axis around but I have to confirm if this actually works as a full axis swap. I'll check and let you know.

**Reply by Paciente8159** [reply]
What kinematics are you using?
It's possible to swap axis (but this lacks extensive testing).

For example the cnc_hal_override.h file you can add the following:

```
#undef AXIS_Z
#undef AXIS_A
#define AXIS_A 2
#define AXIS_Z 3
```

You need to swap the tool axis reference too you should do this also
```
#undef AXIS_TOOL
#define AXIS_TOOL AXIS_A
```
